<launch>

  <arg name="dev_index"     default="0" />
  <arg name="ir_intensity"  default="4" />

  <!-- Stream mode -->
  <arg name="stream_1280x720"   default="0" />
  <arg name="stream_2560x720"   default="1" />
  <arg name="stream_1280x480"   default="2" />
  <arg name="stream_640x480"    default="3" />

  <!-- Stream format -->
  <arg name="stream_mjpg"       default="0" />
  <arg name="stream_yuyv"       default="1" />

  <!-- Depth mode -->
  <arg name="depth_raw"         default="0" />
  <arg name="depth_gray"        default="1" />
  <arg name="depth_colorful"    default="2" />

  <arg name="base_frame"        default="mynteye_link" />
  <arg name="color_frame"       default="mynteye_color" />
  <arg name="depth_frame"       default="mynteye_depth" />
  <arg name="points_frame"      default="mynteye_points_optical" />

  <node name="mynteye_wrapper_node" pkg="mynteye_wrapper" type="mynteye_wrapper_node" output="screen">

    <!-- Camera params -->
    <param name="dev_index"     value="$(arg dev_index)" />
    <param name="framerate"     value="30" />
    <param name="depth_mode"    value="$(arg depth_raw)" />
    <param name="stream_mode"   value="$(arg stream_1280x720)" />
    <param name="stream_format" value="$(arg stream_yuyv)" />
    <param name="state_ae"      value="true" />
    <param name="state_awb"     value="true" />
    <param name="ir_intensity"  value="$(arg ir_intensity)" />

    <!-- Frame ids -->
    <param name="base_frame"    value="$(arg base_frame)" />
    <param name="color_frame"   value="$(arg color_frame)" />
    <param name="depth_frame"   value="$(arg depth_frame)" />
    <param name="points_frame"  value="$(arg points_frame)" />

    <!-- Topic names -->
    <param name="color_topic"   value="mynteye/color" />
    <param name="depth_topic"   value="mynteye/depth" />
    <param name="points_topic"  value="mynteye/points" />

  </node>

  <arg name="pi/2" value="1.5707963267948966" />
  <arg name="optical_rotate" value="0 0 0 -$(arg pi/2) 0 -$(arg pi/2)" />

  <node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="0 0 0 0 0 0 mynteye_link mynteye_color 100" />
  <node pkg="tf" type="static_transform_publisher" name="link2_broadcaster" args="0 0 0 0 0 0 mynteye_link mynteye_depth 100" />
  <node pkg="tf" type="static_transform_publisher" name="link3_broadcaster" args="0 0 0 0 0 0 mynteye_link mynteye_points 100" />
  <node pkg="tf" type="static_transform_publisher" name="link4_broadcaster" args="$(arg optical_rotate) mynteye_link mynteye_color_optical 100" />
  <node pkg="tf" type="static_transform_publisher" name="link5_broadcaster" args="$(arg optical_rotate) mynteye_link mynteye_depth_optical 100" />
  <node pkg="tf" type="static_transform_publisher" name="link6_broadcaster" args="$(arg optical_rotate) mynteye_link mynteye_points_optical 100" />

</launch>
